core/src/main/scala/cats/data/Chain.scala

```diff
@@ -508,11 +500,11 @@ sealed abstract class Chain[+A] {
     val builder = new StringBuilder("Chain(")
     var first = true
 
-    foreach { a =>
+    foreachUntil { a =>
```

Comment:
This is just miscellaneous clean-up, right, since `foreach` isn't used anywhere else?

Reply:
yes. foreach is used exactly once and is private. Just use foreachUntil (or iterator).